common/src/api/external/mod.rs

```diff
@@ -1099,6 +1107,14 @@ impl JsonSchema for Ipv4Net {
 #[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
 pub struct Ipv6Net(pub ipnetwork::Ipv6Network);
 
+impl Ipv6Net {
+    /// Return `true` if this subnetwork is in the IPv6 Unique Local Address
+    /// range defined in RFC 4193, e.g., `fd00:/48`
+    pub fn is_unique_local(&self) -> bool {
+        self.0.network().octets()[0] == 0xfd && self.0.prefix() >= 48
```

Comment:
It's true that a ULA is `fd00:/8` according to the spec. This code really checks that this is a valid ULA prefix _for a VPC_, which is where the /48 comes from. I was trying to avoid magic numbers, but it seems like this is more confusing than helpful. What about putting the mask bits in a constant in `omicron_nexus::defaults`, and then using that at the callsite?

I mean, wherever we call this function, doing something like `ipv6_block.is_unique_local() && ipv6_block.prefix() == defaults::VPC_IPV6_PREFIX_LEN`. Or putting both those into a function, like `is_vpc_prefix()`?

Reply:
I've updated this a bit, by breaking these out into separate methods. `is_unique_local` looks just for `fd` as the first byte. A new method `is_vpc_prefix` checks if something is in the `fd00::/8` _and_ has a `/48` prefix. The new method `is_vpc_subnet` checks that a `/64` subnet is a valid subnetwork of a `Vpc` object, i.e., it shares the same `/48` prefix. Hopefully this clears up some of that confusion.